rr_iarrc/src/urc_controller/urc_controller.cpp

```diff
@@ -0,0 +1,242 @@
+#include <ros/ros.h>
+#include <rr_platform/speed.h>
+#include <rr_platform/steering.h>
+#include <rr_platform/axes.h>
+#include <rr_platform/race_reset.h>
+
+#include <std_msgs/Bool.h>
+#include <std_msgs/Int8.h>
+
+#include <stdlib.h>
+
+const int WAITING_FOR_START = 0;
+const int RUNNING_PLANNER = 1;
+const int FINISHED = 2;
+const int TURNING = 3; 
+const float PI = 3.14; 
+
+std::string startSignal;
+std::string resetSignal;
+
+ros::Publisher steerPub;
+ros::Publisher speedPub;
+
+int state;
+double planSpeed, speed;
+double planSteering, steering;
+
+bool raceStarted;
+bool turnDetected = false;
+bool completed; 
+
+int imu_quadrant; 
+float imu_yaw, initial_yaw, target_yaw;
+
+double left_offset; 
+double right_offset; 
+double turn_speed;
+double left_turn_steering; 
+double right_turn_steering; 
+
+double forward_timeout;
+
+std::string turn_direction = "right";
+
+/*
+// As you turn counterclockwise, the yaw value received from the IMU ranges from [0, 3.14] U [-3.14, 0]
+// This function returns a new yaw value in the range of [0 ,6.28]
+*/
+double setInRange(double yaw_read){
+    double new_yaw; 
+    if(yaw_read < 0){
+        new_yaw = yaw_read + 2*PI; 
+        return new_yaw;
+    }else{
+        return yaw_read;
+    }
+}
+
+/*
+// Checks to see in which quadrant the current yaw reading is in. Turning right while in Q1, and turning left in Q4 are special cases.
+*/
+void checkQuadrant(){
+    if(imu_yaw <= 2*PI && imu_yaw >= 1.5*PI ){
+        imu_quadrant = 4;
+    }else if(imu_yaw >= 0 && imu_yaw <= .5*PI){
+        imu_quadrant = 1;
+    }
+}
+
+// Used to calculate the distance from the target IMU yaw to the current IMU yaw within range of [0 ,6.28]
+bool comp(int a, int b) { 
+    return (a < b); 
+}
+
+float angleDist(float target, float current){
+    float diff = target - current;  
+    float min = std::min({std::abs(diff), std::abs(diff - 2*PI),std::abs(diff + 2*PI)}, comp); 
```

Comment:
comp(a, b) takes in integers? Also, I think "a < b" is the default comparator, in which case min() shouldn't need to have it passed in